### PR TITLE
FEATURE: Make 'Reorder Categories' work with nested categories

### PR DESCRIPTION
**Pull request by udan11**

### CODE REVIEW

File: app/assets/javascripts/discourse/controllers/reorder-categories.js.es6 [hunk 1]
```diff
@@ -31,123 +26,113 @@ export default Controller.extend(ModalFunctionality, Ember.Evented, {
...
+    this.reorder();
   },
 
   actions: {
-    change(cat, e) {
-      let position = parseInt($(e.target).val(), 10);
-      let amount = Math.min(
-        Math.max(position, 0),
+    change(category, event) {
+      let newPosition = parseInt($(event.target).val(), 10);
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/controllers/reorder-categories.js.es6]: Is there a way we could do this without using jQuery?

File: test/javascripts/controllers/reorder-categories-test.js.es6 [hunk 2]
```diff
@@ -30,7 +30,7 @@ QUnit.test("fixIndices set unique position number", function(assert) {
 });
 
 QUnit.test(
-  "fixIndices places subcategories after their parent categories, while maintaining the relative order",
+  "reorder places subcategories after their parent categories, while maintaining the relative order",
```
Comment (ZogStriP) [inline test/javascripts/controllers/reorder-categories-test.js.es6]: We probably want to add another test that ensure sub-sub-categories work properly?

Comment (udan11) [reply test/javascripts/controllers/reorder-categories-test.js.es6]: Amended the current test to include a sub-sub-category.

File: app/assets/javascripts/discourse/controllers/reorder-categories.js.es6 (hunk 1 above)
Comment (udan11) [reply app/assets/javascripts/discourse/controllers/reorder-categories.js.es6]: Looks like a `DOMElement` is already passed. I do not know the reason why jQuery was used, probably just to make the test work.

**Pull request merged**